overview/authentication-flows.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: "Authentication Flows"
+description: "Understanding Civic Auth's flexible OAuth 2.0 authentication methods for maximum security across all application types."
+icon: "arrows-rotate"
+public: true
+---
+
+Civic Auth supports multiple OAuth 2.0 authentication methods to provide maximum security across different application architectures and use cases.
+
+## Authentication Methods
+
+<Tabs>
+  <Tab title="PKCE Only">
+    **Best for:** Public clients (SPAs, mobile apps) where secrets cannot be securely stored.
+
+    ### How it Works
+    - Uses Proof Key for Code Exchange (PKCE) for security
+    - Default method for public clients
+    - Code verifier/challenge mechanism provides security
+    - No client secret required or used
+
+    ### Security Benefits
+    - ✅ No sensitive credentials to manage client-side
+    - ✅ Dynamic verification for each authentication request
+    - ✅ Industry standard for public clients
+    - ✅ Prevents authorization code interception attacks
+
+    ### Configuration
+    ```javascript
+    const config = {
+      clientId: "your_client_id",
+      // PKCE enabled by default, no client secret needed
+      redirectUrl: "https://yourapp.com/callback"
+    }
+    ```
+  </Tab>
+
+  <Tab title="Client Secret Only">
+    **Best for:** Traditional server-side applications requiring OAuth 2.0 compatibility.
```

Comment:
I would point out that this is should only be used for clients that do not support PKCE